tests/TestCase.php

```diff
@@ -114,13 +115,25 @@ protected function clearDirectory($dirPath, $exceptPaths = []): void
         }
     }
 
-    protected function generateRequest($type, $uri, $data = [], $pathParams = [], $headers = [], $routeConditions = [], $controllerMethod = 'test'): Request
-    {
+    protected function generateRequest(
+        $type,
+        $uri,
+        $data = [],
+        $pathParams = [],
+        $headers = [],
+        $routeConditions = [],
+        $controllerMethod = 'test',
+        $isInvokeController = false,
+    ): Request {
         $request = $this->getBaseRequest($type, $uri, $data, $pathParams, $headers);
 
-        return $request->setRouteResolver(function () use ($uri, $request, $controllerMethod, $routeConditions) {
+        return $request->setRouteResolver(function () use ($isInvokeController, $uri, $request, $controllerMethod, $routeConditions) {
+            $action = $isInvokeController
+                ? TestInvokableController::class . '@__invoke'
```

Comment:
for invokable controller the action should be a controller name only

```suggestion
                ? TestInvokableController::class
```